integrations/newrelic/newrelic_integration/core/utils.py

```diff
@@ -0,0 +1,27 @@
+import httpx
+import jinja2
+from loguru import logger
+from port_ocean.context.ocean import ocean
+
+
+async def render_query(query_template: str, **kwargs) -> str:
+    template = jinja2.Template(query_template, enable_async=True)
+    return await template.render_async(
+        **kwargs,
+    )
+
+
+async def send_graph_api_request(query: str, **log_fields) -> dict:
+    async with httpx.AsyncClient() as client:
+        logger.debug("Sending graph api request", **log_fields)
```

Comment:
If you are creating a new instance every time you wont leverage the connection reuse feature 

I think that in 0.1.2 i have a context named http that will help you with this

Reply:
talked offline, implemented my own solution